core/src/test/scala/unit/kafka/KafkaConfigTest.scala

```diff
@@ -41,6 +41,23 @@ class KafkaConfigTest {
   @AfterEach
   def tearDown(): Unit = Exit.resetExitProcedure()
 
+  @Test
+  def testRequiredProperties(): Unit = {
+    // process.roles
+    val properties = new Properties()
+    assertBadConfigContainingMessage(properties,
+      "Missing required configuration \"process.roles\" which has no default value.")
+
+    // node.id
+    properties.put(KRaftConfigs.PROCESS_ROLES_CONFIG, "controller")
+    assertBadConfigContainingMessage(properties,
+      "Missing required configuration \"node.id\" which has no default value.")
+
+    properties.put(KRaftConfigs.NODE_ID_CONFIG, -1)
+    assertBadConfigContainingMessage(properties,
+      "Invalid value -1 for configuration node.id: Value must be at least 0")
```

Comment:
Can we add the case where it works to make sure we don't have any other required configs?